locale.c

```diff
@@ -3701,14 +3653,33 @@ Perl_init_i18nl10n(pTHX_ int printwarn)
             /* For Windows, we also try the system default locale before "C".
              * (If there exists a Windows without LC_ALL we skip this because
              * it gets too complicated.  For those, the "C" is the next
-             * fallback possibility).  The "" is the same as the 0th element of
-             * the array, but the code at the loop above knows to treat it
-             * differently when not the 0th */
-            trial_locales[trial_locales_count++] = (trial_locales_struct) {
-                .trial_locale = "",
-                .fallback_desc = "the system default locale",
-                .fallback_name = "", /* overwritten at the top of the loop */
-            };
+             * fallback possibility). */
+            {
+                const char *system_default_locale = NULL;
+
+                /* Note that this may change the locale, but we are going to do
+                 * that anyway */
+                system_default_locale = setlocale_c(LC_ALL, "");
```

Comment:
As I read this code, I think it would be clearer if the declaration and initialization were combined.